Feature/warn no payment pointer

praghavaji opened the pull request.

Commit:
CWP-128: Testing tweaks

Commit:
Merge branch 'develop' of github.com:coilhq/coil-wordpress-plugin into develop

Commit:
COISUP-6: Admin notice when settings are saved with no payment pointer

paulgibbs commented on assets/js/admin/admin-notices.js:
```diff
@@ -1,25 +1,48 @@
 	$( document ).ready(function() {
+
+		activeTabID = $('.nav-tab-wrapper a.nav-tab-active').attr('id');
+		console.log(activeTabID);
```
No debug logging please

paulgibbs commented on tests/cypress/integration/coil/single-post.js:
```diff
@@ -5,8 +5,8 @@ const paymentPointer = 'https://example.com/' + Math.random().toString(36) + '/.
-  beforeEach(() => {
-    cy.logInToWordPress('admin', 'password');
+	  beforeEach(() => {
+	    cy.logInToWordPress('madhavaji', 'WdX2aJ36iFyZbda9');
```
No passwords in files1

paulgibbs commented on includes/settings/functions.php:
```diff
@@ -503,6 +503,39 @@ function admin_welcome_notice() {
+		return;
+	}
+
+	$current_user       = wp_get_current_user();
```
Unused variable

paulgibbs commented on includes/settings/functions.php:
```diff
@@ -503,6 +503,39 @@ function admin_welcome_notice() {
 	<?php
 }
 
+function admin_no_payment_pointer_notice() {
```
This function should at the very least have a PHPDoc comment explaining what it does.

paulgibbs commented on includes/settings/functions.php:
```diff
@@ -503,6 +503,39 @@ function admin_welcome_notice() {
+	}
+	?>
+
+	<div style="display: none;" class="notice coil-no-payment-pointer-notice">
```
Suggestion: is it worth only showing this nag if the user has the ability to edit the payment pointer field? e.g. someone with limited access to wp-admin would see this nag on the screen, but be unable to directly rectify it.

paulgibbs commented on includes/settings/functions.php:
```diff
@@ -503,6 +503,39 @@ function admin_welcome_notice() {
+		<img width="48" height="48" class="coil-no-payment-pointer-notice__icon" src="<?php echo esc_url( plugins_url( 'assets/images/web-mon-icon.svg', COIL__FILE__ ) ); ?>" alt="<?php esc_attr_e( 'Coil', 'coil-web-monetization' ); ?>" />
+		<div class="coil-no-payment-pointer-notice__content">
+			<h3><?php esc_html_e( 'Warning', 'coil-web-monetization' ); ?></h3>
+			<p>You haven't entered a payment pointer. A payment pointer is required to receive payments and for content gating to be recognized.</p>
```
Since you've put the other new string through a translation function, why did you decide not to do so for this one?

iamthomasbarrett committed:
- Restrict pointer notice nag
- Make string translateable
- Add docblocks
- Remove debug strings
- Remove unusued variable

iamthomasbarrett committed:
Remove usernamme and password

iamthomasbarrett committed:
Fix typo
